cloudify_gcp/compute/instance_group.py

```diff
@@ -88,21 +78,33 @@ def delete(self):
             zone=self.zone,
             instanceGroup=self.name).execute()
 
+    @utils.async_operation(relationship=True)
     @check_response
     def add_instance(self, instance_name):
         return self.discovery.instanceGroups().addInstances(
             project=self.project,
             zone=self.zone,
             instanceGroup=self.name,
-            body=self.instance_to_dict(instance_name)).execute()
+            body=instance_to_dict(instance_name)).execute()
 
+    @utils.async_operation(relationship=True)
     @check_response
     def remove_instance(self, instance_name):
         return self.discovery.instanceGroups().removeInstances(
             project=self.project,
             zone=self.zone,
             instanceGroup=self.name,
-            body=self.instance_to_dict(instance_name)).execute()
+            body=instance_to_dict(instance_name)).execute()
+
+
+def instance_to_dict(instance_url):
```

Comment:
This creates the dict of (one :) Instance url for the instanceGroups().{add,remove}Instances endpoints, the to_dict() methods all produce the full resource dict for $type().insert(). 
So this is a separate idea